### ISSUE TITLE
Job Results API Returning 500 Error

### ISSUE THREAD

**Issue by jathanism** [nautobot/nautobot#3978]
### Environment
* Nautobot version (Docker tag too if applicable): v2 beta
* Python version:
* Database platform, version:
* Middleware(s):

### Steps to Reproduce
0. Migrate from a v1.x dataset
1. Fire up the new UI
2. Load the home page
3. Watch the server logs

### Expected Behavior

No errors

### Observed Behavior

Calls to `http://localhost:8080/api/extras/job-results/?depth=1&limit=10` from the UI return a 500 error.

On the server you see:

```python
12:21:11.308 ERROR   django.server :
  "GET /api/extras/job-results/?depth=1&limit=10 HTTP/1.1" 500 199918
12:21:14.472 ERROR   django.request :
  Internal Server Error: /api/extras/job-results/
Traceback (most recent call last):
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/sentry_sdk/integrations/django/views.py", line 67, in sentry_wrapped_callback
    return callback(request, *args, **kwargs)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/jathan/sandbox/src/nautobot/nautobot/core/api/views.py", line 263, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/mixins.py", line 44, in list
    return self.get_paginated_response(serializer.data)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/serializers.py", line 768, in data
    ret = super().data
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/serializers.py", line 253, in data
    self._data = self.to_representation(self.instance)
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/serializers.py", line 686, in to_representation
    return [
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/serializers.py", line 687, in <listcomp>
    self.child.to_representation(item) for item in iterable
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/rest_framework/serializers.py", line 522, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "/Users/jathan/sandbox/src/nautobot/nautobot/core/api/fields.py", line 46, in to_representation
    return OrderedDict([("value", obj), ("label", self._choices[obj])])
KeyError: 'pending'
```

**Comment by bryanculver** [comment]
This was experienced when attempting to load the V2 Home Page but should affect all API responses.

**Comment by gsnider2195** [comment]
This appears to be caused by the status field not being migrated when the JobResult fields were added in #3164
That migration is tracked in #765 but doesn't have its own issue. If the root cause of this error is the data migration, use this issue to do the migration work.

**Comment by bryanculver** [comment]
Closed with #4140
